hibernate-core/src/test/java/org/hibernate/test/component/proxy/ComponentBasicProxyTest.java

```diff
@@ -44,4 +53,28 @@ public void testBasicProxyingWithProtectedMethodCalledInConstructor() {
 			entityManager.remove( adult );
 		} );
 	}
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-12791")
+	public void testOnlyOneProxyClassGenerated() {
+		StandardServiceRegistry ssr = new StandardServiceRegistryBuilder().build();
+
+		try {
+			Metadata metadata = new MetadataSources( ssr ).addAnnotatedClass( Person.class )
+					.getMetadataBuilder().applyBasicType( new DelimitedStringsType() )
+					.build();
+			PersistentClass persistentClass = metadata.getEntityBinding( Person.class.getName() );
+
+			ComponentType componentType1 = (ComponentType) persistentClass.getIdentifierMapper().getType();
+			Object instance1 = componentType1.instantiate( EntityMode.POJO );
+
+			ComponentType componentType2 = (ComponentType) persistentClass.getIdentifierMapper().getType();
+			Object instance2 = componentType2.instantiate( EntityMode.POJO );
+
+			assertEquals( instance1.getClass(), instance2.getClass() );
```

Comment:
Maybe also assert both classes implement `HibernateProxy`, otherwise we're not testing what we want to test.

Reply:
This particular type of proxy does not implement `HibernateProxy`. The ByteBuddy generated classes implement `ProxyInterceptor` but not the Javassist ones.